### ISSUE TITLE
Simplify Dependencies

### ISSUE THREAD

**Issue by xavez** [iRail/hyperRail#180]
I couldn’t get it to run locally yet, but I can already see we can probably simplify a lot of dependencies:

- [ ] Get rid of the ruby dependency by switching sass and compass to node-sass
- [ ] Get rid of the bower dependency by installing all necessary packages via npm

Unfortunately, I cannot code & test it, since nothing works for me yet, otherwise, I’d have made a PR ;).

**Comment by Tjoosten** [comment]
I assign myself to it. 

To @pietercolpaert I got almost a pr ready for this one.

**Comment by xavez** [comment]
@Tjoosten great!

**Comment by pietercolpaert** [comment]
@Tjoosten Sounds great! You're working on making hyperRail work with gulp instead of grunt, right?

**Comment by Tjoosten** [comment]
@pietercolpaert Yes slowly I will demolish grunt and replace it wilt Laravel Elixir that is based on the GulpJs systems

**Comment by xavez** [comment]
@Tjoosten if you need a good example to start with, check out the `gulp` workflow here: https://github.com/getchopstick/chopstick-boilerplate/blob/master/gulpfile.js or here: https://github.com/mono-company/bedrock/blob/master/gulpfile.js. Not 100% relevant but architecturally a good example.

**Comment by Tjoosten** [comment]
@xavez i'm using the default `laravel-elixir` system for this. 

@pietercolpaert can u describe the asset files. What is included in what asset file. Because it's a horrible experience for me reading these grunt files.

**Comment by pietercolpaert** [comment]
@Tjoosten Sorry, I cannot help here. Maybe @bramdevries can?

**Comment by pietercolpaert** [comment]
@Tjoosten Any update? Are you going to be able to PR this or should someone else do this?

**Comment by Haroenv** [comment]
Maybe open a PR with the work already done so there can be collaboration? There are quite a bit of dependencies right now.

**Comment by Tjoosten** [comment]
Will do it later in the night if you don't mind

**Comment by Haroenv** [comment]
Sure, I don't mind

**Comment by AdriVanHoudt** [comment]
I wanted to tackle this (together with updating to a newer angular version and maybe look into other optimizations) but I couldn't get the server running, I have php but the composer thing just didn't work. Is there maybe a possibility to either help me or  change that to make it easier to get started?

**Comment by Bertware** [comment]
What server are you trying to get running? 
Setting this project up is a pain, from the top of my head you need npm, compass, grunt and bower installed. I'll see if I can reduce the dependencies a bit next week if I have the time, wasn't aware of this issue.

**Comment by pietercolpaert** [comment]
I just tried to run it from scratch. You do not really need to bower and grunt. These were the steps taken:
 (mind that you also now need `php-intl`, [composer](http://getcomposer.org) and [npm](https://docs.npmjs.com/getting-started/installing-node) installed):
```
git clone git@github.com:iRail/hyperRail.git
cd hyperRail/
cp .env.example .env
composer install
php artisan key:generate
chmod -R 777 storage/
npm install
php -S localhost:8081 -t public/
```
And your development version of the hyperRail app should be running at http://localhost:8081

I agree that this workflow is way too complex for a simple application like iRail.be and we should work on a better one. All suggestions to move towards that goal are welcome!

**Comment by AdriVanHoudt** [comment]
I didn't do the env copy so maybe that was it :P, let me try again next week. I might do a pr to the readme first to make the install step clearer 😊

**Comment by Bertware** [comment]
@pietercolpaert Don't forget what's installed globally on your computer already ;) Had to install some tools like compass which weren't installed globally for me yet.

**Comment by pietercolpaert** [comment]
@AdriVanHoudt Thanks! Looking forward!

@Bertware the client-side script builds are included in the repository, so it’s optional to get that up and running... Yet indeed, documentation can be better
